online/src/app/edit-product-form/edit-product-form.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+import { Product } from '../products';
+import { ProductService } from '../services/product.service';
+import { Location } from '@angular/common'
+@Component({
+  selector: 'app-edit-product-form',
+  templateUrl: './edit-product-form.component.html',
+  styleUrls: ['./edit-product-form.component.scss']
+})
+export class EditProductFormComponent implements OnInit {
+  editForm: FormGroup | undefined;
+  product: Product | undefined;
+  constructor(private productService: ProductService, private route: ActivatedRoute, private formBuilder: FormBuilder,
+    private location: Location) { }
+
+  ngOnInit(): void {
+    const id = this.route.snapshot.params['id'];
+    this.productService.getProductById(id).subscribe((item: Product) => {
+      this.product = item;
+      this.editForm = this.formBuilder.group({
+        name: [this.editForm?.value.name, [Validators.required, Validators.minLength(3), Validators.maxLength(100)]],
+        category: [this.product?.category, [Validators.required, Validators.minLength(3), Validators.maxLength(100)]],
+        price: [this.product?.price, [Validators.required, Validators.min(0), Validators.max(10000), Validators.pattern("^[1-9][0-9]*$")]],
+        image: [this.product?.image, [Validators.required, Validators.minLength(20), Validators.maxLength(1000)]],
+        description: [this.product?.description, [Validators.required, Validators.minLength(20), Validators.maxLength(1000)]]
+      });
+      this.editForm.valueChanges.subscribe(console.log);
+    });
+  }
+
+  updateFieldsOfProduct() {
+    if (this.product) {
+      if (!this.editForm?.invalid) {
+        this.product = {
```

Comment:
You can use this.editForm.value and it'll give you the whole object.

Reply:
Thanks for the suggestion. I changed it to this.product = {
          id: this.product.id,
          ...this.editForm.value
        } as I need the ID as well.